src/c++/perf_analyzer/request_rate_manager.cc

```diff
@@ -247,6 +247,23 @@ RequestRateManager::Infer(
   do {
     // Should wait till main thread signals execution start
     if (!execute_) {
+      if (on_sequence_model_) {
+        // Finish off all the ongoing sequences for graceful exit
+        for (size_t i = thread_config->id_; i < sequence_stat_.size();
+             i += thread_config->stride_) {
+          std::lock_guard<std::mutex> guard(sequence_stat_[i]->mtx_);
+          sequence_stat_[i]->paused_ = true;
+          if (sequence_stat_[i]->remaining_queries_ != 0) {
+            ctx->options_->sequence_start_ = false;
```

Comment:
There's two places where these same/similar lines are called. Can this be refactored into a separate function?

Reply:
That's the refactor story. I intentionally left it like this for now to make it easier to review.